lib/system/assertions.nim

```diff
@@ -41,48 +47,54 @@ template assertImpl(cond: bool, msg: string, expr: string, enabled: static[bool]
         failedAssertImpl(ploc & " `" & expr & "` " & msg)
 
 template assert*(cond: untyped, msg = "") =
-  ## Raises ``AssertionDefect`` with `msg` if `cond` is false. Note
-  ## that ``AssertionDefect`` is hidden from the effect system, so it doesn't
-  ## produce ``{.raises: [AssertionDefect].}``. This exception is only supposed
+  ## Raises `AssertionDefect` with `msg` if `cond` is false. Note
+  ## that `AssertionDefect` is hidden from the effect system, so it doesn't
+  ## produce `{.raises: [AssertionDefect].}`. This exception is only supposed
   ## to be caught by unit testing frameworks.
   ##
-  ## The compiler may not generate any code at all for ``assert`` if it is
-  ## advised to do so through the ``-d:danger`` or ``--assertions:off``
+  ## The compiler will not generate any code at all for `assert` if it is
+  ## advised to do so through the `-d:danger` or `--assertions:off`
   ## `command line switches <nimc.html#compiler-usage-commandminusline-switches>`_.
   ##
+  ## **Example:**
```

Comment:
did a runnableExamples not work here?

Reply:
`runnableExamples` doesn't for me, it gives the following error message:
```
Error: ambiguous call; both assertions.assert(cond: untyped, msg: string) [template declared in ~/.choosenim/toolchains/nim-#devel/lib/system/assertions.nim(43, 10)] and assertions.assert(cond: untyped, msg: string) [template declared in ~/Nim/lib/system/assertions.nim(51, 10)] match for: (, string)
```
Or is this just a local issue?

Reply:
you need `--lib:lib` then it works.

we can add the runnableExamples in a next PR

Reply:
I'd like to do that in this PR :)

Do I need to add that command line option somewhere or will it just work? I tried `nim doc --lib:.. assertions.nim` locally and it works.

Reply:
CI expects nim to be "in the right place" wrt sources, ie:
```nim
$nimroot/lib/system.nnim
$nimroot/bin/nim
```
so `--lib:lib` isn't needed; it's only needed if a user is calling this using an external nim eg: `nim doc --lib:lib lib/system/assertions.nim`

likewise with `nim doc lib/system.nim` which will fail (with `semtypes.nim(2055, 12) `c.graph.sysTypes[tySequence] == nil`) if nim is external (adding --lib:lib would work)